### DISCUSSION TITLE
spec: profiles should mandate versions

### DISCUSSION THREAD

**Discussion by wkalt** [foxglove/mcap#1548] [General]
The meaning of a profile could potentially evolve with time. To counteract that, it would be useful for profiles to specify a version.

I think the only reasonable way to do this with the current spec would be to require a specifically-named metadata record. If we roll a V1 at some point we could consider attaching arbitrary k/v data to the Header record where this could potentially live.

**Comment by MichaelOrlov** [comment]
What if we will include profile version in the profile name?

For instance similar as in https://github.com/foxglove/mcap/blob/main/docs/specification/appendix.md#ros2: 

**ROS2v1**
The `ros2` profile describes how to create MCAP files for [ROS 2](https://docs.ros.org/).

**Header**
- `profile`: MUST be `ros2v1`
